app/domain/authentication/authn_jwt/fetch_token_from_credentials.rb

```diff
@@ -0,0 +1,22 @@
+module Authentication
+  module AuthnJwt
+
+    FetchTokenFromCredentials ||= CommandClass.new(
+      dependencies: {
+        decoded_credentials: Authentication::Jwt::DecodedCredentials
+      },
+      inputs: %i[authentication_parameters]
+    ) do
+
+      def call
+        decoded_credentials(@authentication_parameters).jwt
+      end
+
+      private
+
+      def decoded_credentials(authentication_parameters)
+        @decoded_credentials.new(authentication_parameters.credentials)
+      end
```

Comment:
please change to this:
```suggestion
      def call
        decode_credentials
        extracted_token_from_credentials
      end

      private

      def decode_credentials
        decoded_credentials
      end

      def extracted_token_from_credentials
        decoded_credentials.jwt
      end

      def decoded_credentials
        @decoded_credentials_class.new(@authentication_parameters.credentials)
      end
```

Reply:
explanation:
1. the call function should not have ruby code but it should tell the story
2. we don't need to pass the `@authentication_parameters` object to the private method as it can access it by itself. this way we can also memoize the object.